e2e_playwright/custom_components/popular_components_test.py

```diff
@@ -14,13 +14,16 @@
 
 from playwright.sync_api import Page, expect
 
+from e2e_playwright.conftest import wait_for_app_run
+
 
 def _select_component(app: Page, component: str):
     selectbox_input = app.get_by_test_id("stSelectbox").locator("input")
 
     # Type an option (defined in the test app):
     selectbox_input.type(component)
     selectbox_input.press("Enter")
+    wait_for_app_run(app)
 
 
 def _expect_no_exception(app: Page):
```

Comment:
nit: Maybe we can add another `except` to ensure that a custom component is actually rendered?

Reply:
Added in this extracted PR: https://github.com/streamlit/streamlit/pull/8617